module/os/linux/zfs/zpl_super.c

```diff
@@ -187,10 +187,12 @@ __zpl_show_devname(struct seq_file *seq, zfsvfs_t *zfsvfs)
 {
 	char *fsname;
 
+	ZFS_ENTER(zfsvfs);
 	fsname = kmem_alloc(ZFS_MAX_DATASET_NAME_LEN, KM_SLEEP);
 	dmu_objset_name(zfsvfs->z_os, fsname);
 	seq_puts(seq, fsname);
```

Comment:
wouldn't it be better to reduce the lock to just the dmu_objset_name() function call?

Reply:
That would potentially leak memory. ZFS_ENTER has a return in it if the file system is unmounted.

Reply:
OK, sorry for the noise